### DISCUSSION TITLE
Enhancement on top of file based routing

### DISCUSSION THREAD

**Discussion by gulshan** [nitrojs/nitro#2325] [Ideas]
Sometimes I want to keep related codes together, rather than scattering them over multiple files. But to use the current file-based routing, I have to multiple files. I would request some helper methods which may help in this regard.
- Let us export event handler named after HTTP methods. It will allow this-
```typescript
// server/routes/foo.ts
export const get = eventHandler((event) => {
    return "GET request";
});

export const post = eventHandler((event) => {
    return "POST request";
});
```
instead of this-
```typescript
// server/routes/foo.get.ts
export default eventHandler((event) => {
    return "GET request";
});

// server/routes/foo.post.ts
export default eventHandler((event) => {
    return "POST request";
});
```

- Also let us define inline routes with a helper method like `createRoute`. It will allow this-
```typescript
// server/routes/foo.ts
export default eventHandler((event) => {
    return "foo handler";
});

export const bar = defineRoute(eventHandler((event) => {
    return "bar handler";
}))
```

instead of this-
```typescript
// server/routes/foo/index.ts
export default eventHandler((event) => {
    return "foo handler";
});

// server/routes/foo/bar.ts
export default eventHandler((event) => {
    return "bar handler";
});
```
- These two can be combined to something like this-
```typescript
export const baz = defineRoute({
    get: eventHandler((event) => {
        return "GET request";
    }),
    post: eventHandler((event) => {
        return "POST request";
    }),    
})
```
These will give developers some flexibility while defining routes.

**Comment by pi0** [comment]
This is something we certainly have to support at some point. Thanks for feedback.

**Reply by gulshan** [reply]
Pinging for a revisit.

**Reply by pawmanaloto** [reply]
Pinging this as well, we have so many endpoints and putting them per file just so hard to maintain in the long run.

**Reply by adelinerohr** [reply]
Following up on this as well
